src/Sylius/Bundle/CoreBundle/Fixture/Factory/CatalogPromotionExampleFactory.php

```diff
@@ -110,6 +94,7 @@ public function create(array $options = []): CatalogPromotionInterface
             }
         }
 
+        $this->catalogPromotionRepository->add($catalogPromotion);
```

Comment:
We definitely should not persist the entity in the **factory** class which should only be responsible for creation. Especially, as fixture already persist it after creation :dancer: Perhaps we should move `$this->allCatalogPromotionsProcessor->process();` to listener implementing `AfterFixtureListenerInterface`

Reply:
And, btw, it's already done [here](https://github.com/Sylius/Sylius/pull/13579/files#diff-6120934584af520c17ad611a2af6f0743108f3db475350768ce5b0f84ab03ae2R23) :smile: